src/main/java/io/snowdrop/reporting/weekreport/WeekReportEndpoint.java

```diff
@@ -0,0 +1,150 @@
+    String repoName;
+
+    @Inject
+    GitHubClient client;
+
+    @Inject
+    GithubReportingService service;
+
+    private void populate(Date startTime, Date endTime) {
+        Set<PullRequest> pullRequests = service.getPullRequestCollector().getPullRequests().filter(p -> p.isActiveDuring(startTime, endTime))
+                .collect(Collectors.toSet());
+        Set<Issue> issues = service.getIssueCollector().getIssues().filter(i -> i.isActiveDuring(startTime, endTime)).collect(Collectors.toSet());
+
+        Set<RepositoryWork> workItems = service.getRepositoryCollector().getUsers().stream().flatMap(u -> {
+            Map<String, Set<Issue>> userIssues = issues.stream().filter(i -> u.equals(i.getAssignee()))
+                    .collect(Collectors.groupingBy(i -> i.getRepository(), Collectors.toSet()));
+            Map<String, Set<PullRequest>> userPullRequests = pullRequests.stream().filter(p -> u.equals(p.getCreator()))
+                    .collect(Collectors.groupingBy(p -> p.getRepository(), Collectors.toSet()));
+            return Stream.concat(userIssues.keySet().stream(), userPullRequests.keySet().stream())
+                    .distinct()
+                    .map(r -> new RepositoryWork(Github.user(r), Github.repo(r), u,
+                            userIssues.getOrDefault(r, Collections.emptySet()),
+                            userPullRequests.getOrDefault(r, Collections.emptySet())));
+        }).collect(Collectors.toSet());
+    }
+
+    @GET
+    @Path("/generate")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public String createReport(@QueryParam("startTime") String startTimeString, @QueryParam("endTime") String endTimeString) {
+        String mdText;
+        String weekNumber;
+        IssueService issueService = new IssueService(client);
+        try {
+            Date startTime = startTimeString != null ? DF.parse(startTimeString)
+                    : Date.from(service.getPullRequestCollector().getStartTime().toInstant());
+            Date endTime = endTimeString != null ? DF.parse(endTimeString)
+                    : Date.from(service.getPullRequestCollector().getEndTime().toInstant());
+            weekNumber = WEEK_YEAR_FORMAT.format(endTime);
+            LOGGER.debug("week number: " + weekNumber);
+            populate(startTime, endTime);
+            mdText = WeeklyDevelopmentReportImpl.build(startTime, endTime, users).buildWeeklyReport(Repository.findAll().list());
+            LOGGER.info("weekly_development: " + mdText);
+            updateGithubIssue(mdText, "weekly_development_" + weekNumber, "report", issueService, repoOrganization, repoName);
+            mdText = DevelopmentReportImpl.build(startTime, endTime, users).buildWeeklyReport(Repository.findAll().list());
+            LOGGER.info("dev: " + mdText);
+            updateGithubIssue(mdText, "dev_" + weekNumber, "report", issueService, repoOrganization, repoName);
```

Comment:
I think that we should have the possibility to first review the report populated and that we propose a separate action form button to create the github issue or discussion if we click on it. WDYT ?

Reply:
I think it won't be straightforward because it would mean having a markdown parser embedded in the application. What actions do you think we would do before actually adding the report to it's destination?

Reply:
I like this idea a lot! 

It will make development so much easier and it will give people the chance to see what is going in, without messing with the actual repo. 

So, I'd suggest we turn this suggestion into a feature request that we can implement next.

Reply:
Done, the issue is #62